test/integration/tree_test.go

```diff
@@ -792,6 +792,225 @@ func TestTree(t *testing.T) {
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오랑 2 네이버3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, d2.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 6}, []int{1, 0, 0, 7}, nil, 0)
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 5}, []int{1, 0, 0, 6}, nil, 0)
+
+			return nil
+		}))
+
+		assert.NoError(t, c2.Sync(ctx))
+		assert.NoError(t, c1.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이버3</n></p></c></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이버3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, d1.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 9}, []int{1, 0, 0, 10}, nil, 0)
+
+			return nil
+		}))
+
+		assert.NoError(t, c1.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이3</n></p></c></r>", d1.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		c3 := activateSingleClient(t)
+		d3 := document.New(helper.TestDocKey(t))
+		assert.NoError(t, c3.Attach(ctx, d3))
+
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이3</n></p></c></r>", d3.Root().GetTree("t").ToXML())
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이3</n></p></c></r>", d3.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카카오2 네이3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, d3.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 4}, []int{1, 0, 0, 5}, nil, 0)
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 3}, []int{1, 0, 0, 4}, nil, 0)
+
+			return nil
+		}))
+
+		assert.NoError(t, c3.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카2 네이3</n></p></c></r>", d3.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 카2 네이3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, d3.Update(func(root *json.Object, p *presence.Presence) error {
+			root.GetTree("t").EditByPath([]int{1, 0, 0, 2}, []int{1, 0, 0, 3}, nil, 0)
+
+			return nil
+		}))
+
+		assert.NoError(t, c3.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 2 네이3</n></p></c></r>", d3.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 2 네이3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, c3.Deactivate(context.Background()))
+		assert.NoError(t, c3.Close())
+	})
+
```

Comment:
I remember that we will be working on simplifying the test code. Will you work on the next PR?

Reply:
Of course, I will simplify it and reflect it in the next PR.